.github/workflows/fw-test-emu.yml

```diff
@@ -47,19 +62,45 @@ jobs:
 
       - name: Checkout repo
         uses: actions/checkout@v3
+        with:
+          submodules: 'true'
 
       - name: Pull dpe submodule
         run: |
           git submodule update --init dpe
 
-      - name: Build firmware
-        run: |
-          mkdir /tmp/caliptra-test-firmware
-          cargo run -p caliptra-builder -- --all_elfs /tmp/caliptra-test-firmware
+      - name: Extract Versioned ROM and FMC/RT
+        uses: actions/download-artifact@v4
+        with:
+          # defined in versioned-checkout.yml, files will be moved to /tmp/caliptra
+          # there is no hw version definition here because we are running in the emulator, so assume latest
+          name: caliptra-unified-${{ inputs.rom-ref }}-${{ inputs.fw-ref }}-latest-${{ inputs.artifact-suffix }}
+          path: /tmp/caliptra-unified
 
+      - name: Update rustup
+        run: |
+          rustup toolchain install nightly-2023-04-15-x86_64-unknown-linux-gnu
+          rustup default nightly-2023-04-15-x86_64-unknown-linux-gnu
+          rustup target add riscv32imc-unknown-none-elf --toolchain nightly-2023-04-15-x86_64-unknown-linux-gnu
+          rustup component add rust-src --toolchain nightly-2023-04-15-x86_64-unknown-linux-gnu
+          rustup show
+          cargo version
+          ls -la /tmp/
+          df -hi
+      
       - name: Run tests
         run: |
-          export CALIPTRA_PREBUILT_FW_DIR=/tmp/caliptra-test-firmware
+          export CALIPTRA_PREBUILT_FW_DIR=/tmp/caliptra-unified/
+          export CALIPTRA_PREBUILT_ROM_BIN=/tmp/caliptra-unified/caliptra-rom.bin
+          export RUST_BACKTRACE=full
+
+          rustup toolchain install nightly-2023-04-15-x86_64-unknown-linux-gnu
+          rustup default nightly-2023-04-15-x86_64-unknown-linux-gnu
+          rustup target add riscv32imc-unknown-none-elf --toolchain nightly-2023-04-15-x86_64-unknown-linux-gnu
+          rustup component add rust-src --toolchain nightly-2023-04-15-x86_64-unknown-linux-gnu
```

Comment:
I opened https://github.com/chipsalliance/caliptra-sw/pull/2138 to fix the compilation error that @MDr164 mentioned this morning.

I believe the toolchain was set by `rust-toolchain.toml` and then the tests change directories and use the global toolchain. 

My PR changes this snippet to make the toolchain match `rust-toolchain.toml`.

Can you clarify the usage of `nightly-2023-04-15-x86_64-unknown-linux-gnu` here?